### PR TITLE
fix(cors): allow all valid regex

### PR DESCRIPTION
**Pull request by NicolasGeraud**
**Issue**

https://github.com/gravitee-io/issues/issues/5611

**Description**
a valid regex has to be accepted in the allowed origin

### CODE REVIEW

File: gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/ApiService_UpdateTest.java [hunk 1]
```diff
@@ -701,7 +702,7 @@ public void shouldNotChangeLifecycleStateFromCreatedInReview() throws TechnicalE
     @Test(expected = AllowOriginNotAllowedException.class)
     public void shouldHaveAllowOriginNotAllowed() throws TechnicalException {
         prepareUpdate();
-        existingApi.getProxy().getCors().getAccessControlAllowOrigin().add("/test^");
+        existingApi.getProxy().getCors().getAccessControlAllowOrigin().add("(/test^");
```
Comment (brasseld) [inline gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/ApiService_UpdateTest.java:705]: What is the reason for this change ?

**Review by brasseld — CHANGES_REQUESTED** [review]

File: gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/ApiService_UpdateTest.java (hunk 1 above)
Comment (NicolasGeraud) [reply gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/ApiService_UpdateTest.java:705]: to test an invalid regex.
It's not possible to test that a regex created by a customer is a valid origin. We can just test if the origin is a valid address or if it's a valid regex (even if the regex represents an invalid origin).
